qiskit/result/distributions/probability.py

```diff
@@ -81,7 +79,7 @@ def binary_probabilities(self, num_bits=None):
             dict: A dictionary where the keys are binary strings in the format
                 ``"0110"``
         """
-        n = len(bin(max(self.keys(), default=0))) - 2 if num_bits is None else num_bits
+        n = num_bits or self._num_bits
```

Comment:
Would
```suggestion
        n = num_bits if num_bits is not None else self._num_bits
```
be safer since `num_bits=0` would evaluate to False and falsely use `self._num_bits`?

Reply:
Thanks. You are right.

Reply:
done